server/views/pages/approve/cases.njk

```diff
@@ -48,7 +48,22 @@
                 "data-sort-value": case.probationPractitioner
             },
             html: '<a href="/licence/approve/id/' + case.licenceId + '/probation-practitioner" class="govuk-link">' + case.probationPractitioner.name + '</a>'
-        },
+        }
+    ] %}
+
+    {% if hardStopEnabled and approvalNeededView %}
+        {% set caseData = (caseData.push(
+            {
+                attributes: {
+                    id: 'approved-by-' + loop.index,
+                    "data-sort-value": case.submittedBy
+                },
+                text: case.submittedBy
+            }
+        ), caseData) %}
+    {% endif %}
+
+    {% set caseData = (caseData.push(
```

Comment:
If there's no other way to accomplish conditional rendering of a column part-way through, I understand that this may be our only approach. But damn I'm really not a fan of having to push columns in like this, especially when the last column is always added.

Is there perhaps a way to insert at a certain index instead? Or is this a pattern we use elsewhere?

Reply:
Updated it with splice.